src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,141 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final Object[] EMPTY_LIST = {};
+
+    private Object[] listElements;
+    private int size = 0;
+
+    public ArrayList(int initialCapacity) {
+        if (initialCapacity > 0) {
+            this.listElements = new Object[initialCapacity];
+        } else if (initialCapacity == 0) {
+            this.listElements = EMPTY_LIST;
+        } else {
+            throw new IllegalArgumentException("Illegal Capacity: " + initialCapacity);
+        }
+    }
+
+    public ArrayList() {
+        this(DEFAULT_CAPACITY);
+    }
+
+    public ArrayList(Collection<? extends T> collection) {
+        this(collection.size());
+
+        for (T collectionElement : collection) {
+            add(collectionElement);
+        }
+    }
+
```

Comment:
the same. Let's remove this constructor and leave the constructor without parameters ;)